src/main/java/com/fundynamic/d2tm/game/gui/BuyStuffGuiElement.java

```diff
@@ -0,0 +1,168 @@
+import com.fundynamic.d2tm.game.controls.Mouse;
+import com.fundynamic.d2tm.game.entities.Rectangle;
+import com.fundynamic.d2tm.game.rendering.RenderQueue;
+import com.fundynamic.d2tm.graphics.ImageRepository;
+import com.fundynamic.d2tm.math.Vector2D;
+import org.newdawn.slick.Color;
+import org.newdawn.slick.Graphics;
+import org.newdawn.slick.Image;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BuyStuffGuiElement implements Renderable, Updateable {
+
+    private Rectangle rectangle;
+
+    private BuyIconList buyIconList = new BuyIconList();
+
+    private final int widthOfIcon;
+    private final int heightOfIcon;
+
+    private Mouse mouse;
+
+    public BuyStuffGuiElement(int x, int y, Mouse mouse) {
+        this.mouse = mouse;
+        // TODO: Do this properly, because a getInstance is a very ugly way to get this dependency here.
+        ImageRepository instance = ImageRepository.getInstance();
+        List<Image> images = new ArrayList<>();
+        images.add(instance.loadAndCache("ui/icons/icon_1slab.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_4slab.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_barracks.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_heavyfactory.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_hightech.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_ix.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_lightfactory.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_palace.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_radar.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_refinery.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_repair.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_rturret.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_silo.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_starport.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_turret.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_wall.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_windtrap.bmp"));
+        images.add(instance.loadAndCache("ui/icons/icon_wor.bmp"));
+
+        // 3 icons in a row. We assume the dimensions of the first one
+        widthOfIcon = images.get(0).getWidth();
+        heightOfIcon = images.get(0).getHeight();
+
+        // the height of this window is:
+        // 26 pixels for the 'top' (ie, the title of the thing, Const Yard, etc)
+        // then we add the height + 1 pixel room (in between the icons)
+        int rows = 1 + (int) Math.floor(images.size() / 4);
+
+        int heightOfWindow = 26 + (rows * heightOfIcon) + rows + 1;
+        int widthOfWindow =  1 + (4 * widthOfIcon) + 3 + 1; // 1 pixel left border + 4 icons + 3 times in between a pixel + 1 pixel for border at the right
```

Comment:
4 is a magic number in a few statements in this class. It would improve readability if it's extracted to a const `ICONS_PER_ROW` (or something like that)